packer/files/conf-from-user.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python3
+
+import urllib.request
+import json
+import subprocess
+import shutil
+import shlex
+import yaml
+import argparse
+import email
+from pathlib import Path
+
+HOME_DIR='/home/centos/'
+USER='centos'
+dry_run=None
+verbose=None
+
+"""
+{
+"version": "master,5.1",
+"ips":["127.0.0.1", "127.0.0.2"]
+}
+"""
+
+def run(template_cmd, shell=False):
+    cmd = template_cmd.format(_HOME=HOME_DIR, _USER=USER)
+    if dry_run:
+        print(cmd)
+        return
+    if verbose:
+        print(cmd)
+    if not shell:
+        cmd = shlex.split(cmd)
+    try:
+        res = subprocess.check_output(cmd, shell=shell)
+        if verbose:
+            print(res)
+        return res
+    except Exception as e:
+        print("Error while running:")
+        print(cmd,end =" ")
+        print(e.output)
+        raise
+
+def parse_message(response):
+        r = response.read()
+        msg = email.message_from_bytes(r)
+        res = {}
+        if msg.is_multipart():
+            for part in msg.walk():
+                obj = None
+                filename = part.get_filename()
+                base_filename =  filename.split('.')[0] if filename else None
+                # if base_filename and base_filename in ['scylla_server', 'scylla-monitoring', 'env']:
+                if part.get_content_type().startswith('application/') or part.get_content_type().startswith('text/plain'):
+                    type = part.get_content_type().split('/')[1].strip(' \t\n\r')
+                    pyload = None
+                    if type.endswith('yml') or type.endswith('yaml'):
+                        pyload = part.get_payload(decode='utf-8')
+                        obj = yaml.safe_load(pyload)
+                    elif type.endswith('json'):
+                        pyload = part.get_payload(decode='utf-8')
+                        obj = json.load(pyload)
+                    else:
+                        obj = part.get_payload(decode=True).decode('utf-8')
+                if obj:
+                    res[base_filename] = {'obj' : obj, 'filename': filename, 'payload' : pyload}
+        else:
+            print("message is not a multipart, ignoring")
+        return res
+
+def get_parsed_message(url, headers={}):
+    try:
+        req = urllib.request.Request(url, headers=headers)
+        response = urllib.request.urlopen(req)
+        return parse_message(response)
+    except Exception as err:
+        print(f"Unexpected {err=}, {type(err)=}")
+        return None
+
+def getAWSData(args):
+    url = 'http://169.254.169.254/latest/user-data' if args.address == "" else args.address 
```

Comment:
this will fail with IMDSv2 - unsure when it'll be the default and enforced as such in AWS.